lib/vtls/darwinssl.c

```diff
@@ -1034,6 +1034,129 @@ CF_INLINE bool is_file(const char *filename)
   return false;
 }
 
+static CURLcode
+set_ssl_version_up_to(struct connectdata *conn, int sockindex,
+                      long ssl_version, long ssl_version_up_to)
+{
+  struct Curl_easy *data = conn->data;
+  struct ssl_connect_data *connssl = &conn->ssl[sockindex];
+
+  switch(ssl_version_up_to) {
+    case CURL_SSLVERSION_OR_UP_TO_NONE:
+      switch(ssl_version) {
+        case CURL_SSLVERSION_TLSv1_0:
+          return set_ssl_version_up_to(conn, sockindex, ssl_version,
+                                       CURL_SSLVERSION_OR_UP_TO_TLSv1_0);
```

Comment:
Why do you recursively call the function?  Would not it be enough to (re)assign ```ssl_version_up_to``` and continue the execution?